### ISSUE TITLE
[YSQL] Read time should NOT be specified for serializable isolation level

### ISSUE THREAD

**Issue by FranckPachot** [yugabyte/yugabyte-db#14284]
Jira Link: [DB-3726](https://yugabyte.atlassian.net/browse/DB-3726)
### Description

With a DELETE and INSERT in the same SERIALIZABLE transaction, in order to move rows from one table to another, I get:
```
ERROR:  Invalid argument: [Invalid argument (yb/tablet/write_query.cc:529): Read time should NOT be specified for serializable isolation level: { read: { physical: 1664650950759987 } local_limit: { physical: 1664650950759987 } global_limit: { physical: 1664650951259988 } in_txn_limit: { physical: 1664650950772110 } serial_no: 0 }, Invalid argument (yb/tablet/write_query.cc:529): Read time should NOT be specified for serializable isolation level: { read: { physical: 1664650950759987 } local_limit: { physical: 1664650950766657 } global_limit: { physical: 1664650951259988 } in_txn_limit: { physical: 1664650950772110 } serial_no: 0 }]
```
Here is the reproducible test case:
```

-- create the origin table:
create table demo as select tenant_id, id, 0 a, 0 b, 0 c
 from generate_series(1,100) tenant_id, generate_series(1,1000) id
;

-- create the destination table:
create table demo2 as 
 select * from demo
 where null is not null --> always false, to read no rows
;
alter table demo2 add primary key (tenant_id, id);

-- set serializable isolation level
set default_transaction_isolation to serializable;

-- INSERT from a DELETE rowset
with deleted_rows as (
 delete from demo
 where tenant_id=42 
 returning *
) insert into demo2 select deleted_rows.* from deleted_rows
;
```

**Comment by FranckPachot** [comment]
Adding another test case from [forum question](https://forum.yugabyte.com/t/invalid-argument-read-time-should-not-be-specified-for-serializable-isolation-level/1836/3?u=franckpachot)
```
/* setup */
SET TRANSACTION ISOLATION level serializable;
CREATE TABLE table_aa (
    id uuid NOT NULL
);
ALTER TABLE ONLY table_aa
    ADD CONSTRAINT "PK_table_aa" PRIMARY KEY (id);
   
CREATE TABLE table_bb (
    id uuid NOT NULL,
    prop character(2) NOT NULL
);

ALTER TABLE ONLY table_bb
    ADD CONSTRAINT "PK_table_bb" PRIMARY KEY (id, prop);
ALTER TABLE ONLY table_bb
    ADD CONSTRAINT "FK_table_bb_table_aa_id" FOREIGN KEY (id) REFERENCES public.table_aa(id) ON DELETE CASCADE;

insert INTO public.table_aa (id) VALUES ('00000000-0000-0000-0000-000000000000');
insert INTO public.table_bb (id, prop) VALUES ('00000000-0000-0000-0000-000000000000', 'AA');
insert INTO public.table_bb (id, prop) VALUES ('00000000-0000-0000-0000-000000000000', 'BB');

/* transaction that throws error */
BEGIN TRANSACTION ISOLATION LEVEL serializable;
SELECT t.id
        FROM (
            SELECT a.id
            FROM table_aa AS a
            WHERE a.id = '00000000-0000-0000-0000-000000000000'
            LIMIT 1
        ) AS t
        LEFT JOIN table_bb AS a0 ON t.id = a0.id;
SAVEPOINT "__EFSavePoint";
UPDATE table_aa SET id = '00000000-0000-0000-0000-000000000000' where id = '00000000-0000-0000-0000-000000000000';
commit;
```

**Comment by m-iancu** [comment]
The commit above fixes the issue in latest: [a0ff61b](https://github.com/yugabyte/yugabyte-db/commit/a0ff61bcfc8aee889172544a03a1a65e781e3e2e). Next steps are backports to 2.16 and 2.18 branches (which are also affected) before we close this.
